python/cudf/cudf/core/index.py

```diff
@@ -443,6 +443,12 @@ def __init__(self, start, stop=None, name=None):
         self._cached_values = None
 
     def __contains__(self, item):
+        if not isinstance(
+            item, tuple(np.sctypes["int"] + np.sctypes["float"] + [int, float])
+        ):
+            return False
+        elif not item % 1 == 0:
+            return False
```

Comment:
I'm confused by this. Can you explain? The test cases you wrote only test `StringIndex`, does this execute?

Reply:
this is from a previous iteration of my tests where the columns weren't named, giving a `RangeIndex`. In that case checking if some column names were `in` the RangeIndex raised a `TypeError`. The mod is just there to make `2.0` work but `2.1` won't even though it's between the bounds. 

Adding a separate test for this.